File: cs/Parking/Car.cs
namespace parking
{
    public class Car : ValueObject<PlateNumber>
    {
        public Car(string plateNumber) : base(new PlateNumber(plateNumber)) { }

        public PlateNumber PlateNumber => value;
    }

    public class PlateNumber : ValueObject<string>
    {
        public PlateNumber(string plateNumber) : base(plateNumber) { }
    }
}

File: ts/src/parking/domain/parkingLot/parking/car.ts
/** @format */

import {ValueObject} from '../../valueObject'

export class Car extends ValueObject<PlateNumber> {
    public constructor(plant: String) {
        super(new PlateNumber(plant))
    }

    public get plant() {
        return this.value
    }
}

export class PlateNumber extends ValueObject<String> {}
